prov/gni/src/gnix_mr.c

```diff
@@ -440,6 +464,10 @@ int gnix_mr_reg(struct fid *fid, const void *buf, size_t len,
 		}
 	}
 
+	reg_addr = ((uint64_t) buf) & ~((1 << GNIX_MR_PAGE_SHIFT) - 1);
+	reg_len = __calculate_length((uint64_t) buf, len,
+			1 << GNIX_MR_PAGE_SHIFT);
```

Comment:
i thought the rounding up/down was going to take in to account the pagesize info stored in the gni memhndl,